java/client/src/main/java/glide/api/models/ClusterValue.java

```diff
@@ -0,0 +1,44 @@
+package glide.api.models;
+
+import java.util.Map;
+
+/**
+ * A union-like type which can store single or bulk value retrieved from Redis.
+ *
+ * @param <T> The wrapped data type
+ */
+public class ClusterValue<T> {
+    private Map<String, T> multiValue = null;
+
+    private T singleValue = null;
+
+    private ClusterValue() {}
+
+    /** Get per-node value. */
+    public Map<String, T> getMultiValue() {
+        assert hasMultiData();
+        return multiValue;
+    }
+
+    /** Get the single value. */
+    public T getSingleValue() {
```

Comment:
javadoc - please mention that hasSingleValue() should be called first to avoid exceptions